lib/aws-manager.js

```diff
@@ -81,142 +82,164 @@ class AwsManager {
+      // Submit request to kill stalled requests
+      stalledKills.push(this.killCancel(region, [], stalledSR));
 
-      allPricingHistory = await this.ec2.describeSpotPriceHistory({
-        StartTime: pricingStartDate,
-        Filters: [
-          {
-            Name: 'product-description',
-            Values: ['Linux/UNIX'],
-          },
-        ],
-      });
-      debug('fetched all pricing history');
-      this.__pricing = this._findMaxPrices(allPricingHistory, this.__availableAZ);
-    } catch (err) {
-      debug('error running a status or pricing promises');
-      debug(err);
-      if (err.stack) {
-        debug(err.stack);
-      }
-      throw err;
-    }
-
-    // NOTE: Dead spot request does not mean that the spot request failed,
-    //       just that it's no longer an open spot request...
+      for (let reservation of response[2].data.Reservations) {
+        for (let instance of reservation.Instances) {
+          let workerType = this.parseKeyPairName(instance.KeyName).workerType;
+          // Maybe use objFilter here
+          let filtered = instance;
+          filtered.Region = region;
+          filtered.WorkerType = workerType;
+          deadState.instances.push(filtered);
+        }
+      };
 
-    // TODO: Probably better to do a single set of describe calls and bucket
-    //       them in this section... hmm, no strong feelings until data
+      deadSpotRequests[region] = [];
+      for (let request of response[3].data.SpotInstanceRequests) {
+        let workerType = this.parseKeyPairName(request.LaunchSpecification.KeyName).workerType;
+        // Maybe use objFilter here
+        let filtered = request;
+        filtered.Region = region;
+        filtered.WorkerType = workerType;
+        deadState.requests.push(filtered);
+      }
 
-    // We don't currently handle this case because it seems to be deprecated in
-    // the docs.  We should be alerted if it begins to be used again
-    if (livingInstances.NextToken && livingInstances.NextToken !== '') {
-      debug('[alert-operator] WARNING!  We have a ' +
-          'DescribeInstances NextToken and arent using it!!!');
-    }
+      // Remember we don't filter these the same way that
+      // we filter the other responses
+      availableAZ[region] = response[4].data.AvailabilityZones.map(x => x.ZoneName);
+      allPricingHistory[region] = this._findMaxPrices(response[5].data, availableAZ[region]);
+    });
 
-    let filteredSpotRequests = this._filterSpotRequests(livingSpotRequests);
+    this.__availableAZ = availableAZ;
+    this.__pricing = allPricingHistory;
+    this.__apiState = apiState;
+    this.__deadState = deadState;
 
-    // We store the last iteration's state
-    this.__previousApiState = this.__apiState;
-    this.__apiState = this._classify(livingInstances, filteredSpotRequests.good);
-    this.__deadState = this._classify(deadInstances, deadSpotRequests);
+    await Promise.all(stalledKills);
```

Comment:
Because async() evaluates async, `stalledKills == []` always.. the place where you push entries to it is never reached before the `_.forEach` is done.. as `_.forEach` is sync...